src/ports/postgres/modules/stats/correlation.py_in

```diff
@@ -291,8 +296,28 @@ def _populate_output_table(schema_madlib, source_table, output_table,
                 {deconstruction_query}
             ) {matrix_subquery}
             USING (column_position)
-            """.format(num_cols=len(col_names), **locals())
+        """
+
+        # Create the output table.
+        create_output_table_query = """
+            CREATE TABLE {output_table} AS
+            {select_deconstruct_query}
+            """.format(**locals()).format(
+                deconstruction_query=deconstruction_query_list[0],
```

Comment:
consider adding a comment to explain why we create the table using the first subquery. I noticed that there is a comment on line 309 but we should make it more clear and move it before the create table query.